### PR TITLE
Use syncronized file saving on Linux 

### PR DESCRIPTION
**Pull request by pyhys**
This is required for file changes to appear on disk, even if the
power is removed soon after writing. Tested with ART tester test
case "Topology discovery check".

Bug reported by Giuseppe.

### CODE REVIEW

File: src/ports/linux/pnal.c [hunk 1]
```diff
@@ -97,6 +97,7 @@ int pnal_save_file (
 
    /* Close file */
    fclose (outputfile);
+   sync();
```
Comment (olbjo) [inline src/ports/linux/pnal.c:100]: Maybe fsync() is better. Syncing/flushing whole file system should probably be avoided.
I think we should rerun test with fsync to check if that works and also compare the timing of  the two

**Review by olbjo — CHANGES_REQUESTED** [review]

File: src/ports/linux/pnal.c (hunk 1 above)
Comment (pyhys) [reply src/ports/linux/pnal.c:100]: Ok, I will try.

Comment (pyhys) [reply src/ports/linux/pnal.c:101]: The `fsync()` approach did unfortunately not work, but I removed the `sync()` from `pnal_clear_file()`.

Comment (olbjo) [reply src/ports/linux/pnal.c:101]: OK

**Review by olbjo — APPROVED** [review]
👍

File: src/ports/linux/pnal.c (hunk 1 above)
Comment (hefloryd) [reply src/ports/linux/pnal.c:101]: It might be worth investigating O_SYNC | O_DIRECT.

Comment (pyhys) [reply src/ports/linux/pnal.c]: `fflush()` and `fsync()` together did the trick.

**Comment by pyhys** [comment]
Changed to use non-buffered kernel API.

**Comment by pyhys** [comment]
Remove redundant `fsync` when using `O_SYNC`

**Pull request merged**